zokrates_core/src/flatten/mod.rs

```diff
@@ -355,13 +355,31 @@ impl Flattener {
 
                 FlatExpression::Identifier(name_1_y)
             },
+            BooleanExpression::Le(box lhs, box rhs) => {
+                let lt = self.flatten_boolean_expression(functions_flattened,
```

Comment:
Wouldn't it be possible to flatten `Or(Lt, Eq)` instead of flattening `Lt` and `Eq` and then adding them?

Reply:
Yes, I just did this so it would be independent from the boolean operator branch, but since that's merged we can use that.

It is actually more efficient to do it this way, since boolean Or uses a multiplication gate but this is guaranteed to be boolean with just an addition.

Reply:
Ok let's keep it like this for now

Reply:
I think it would be clearer to use `Or` in this case. It's a minor performance hit, and we can potentially make `Or` flattening better if needed.